docs/codegen/overview.md

```diff
@@ -4,32 +4,70 @@ Code generation is the main way you can glue [Terramate data](../sharing-data.md
 to other tools, like Terraform.
 
 Different code generation strategies will be provided in the future to support
-each integration scenario in the best way possible. Currently, we support:
+each integration scenario in the best way possible. 
 
-* [HCL generation](./generate-hcl.md)
-* [File generation](./generate-file.md)
+Currently, we support:
 
-Some features are available to all code generation strategies, like:
+* [HCL generation](./generate-hcl.md) with stack [context](#generation-context).
+* [File generation](./generate-file.md) with `root` and `stack` [context](#generation-context).
+
+# Generation Context
+
+The Terramate code generation supports two execution context:
+
+- stack: generates code relative to the stack where it's defined.
+- root: generates code outside of stacks.
+
+The `stack` context gives access to all code generation features, like:
 
 * [Globals](../sharing-data.md#globals)
-* [Metadata](../sharing-data.md#metadata)
+* [All Metadata](../sharing-data.md#metadata)
 * [Functions](../functions.md)
 * [Lets](#lets)
 * [Assertions](#assertions)
 
+But the `root` context gives access to:
+
+* [Project Metadata](../sharing-data.md#project-metadata)
+* [Functions](../functions.md)
+* [Lets](#lets)
+
+If not specified otherwise, the default generation context is `stack`.
+For the `generate_hcl` there's no way to change the `context` and it's always
+of type `stack` but the `generate_file` has a `context` attribute which you can explicit change it to `root`. Example:
```

Comment:
```suggestion
The `generate_hcl` block doesn't support changing the `context`, it will always be
of type `stack`. The `generate_file` block supports the `context` attribute which you can explicit change to `root`.
Example:
```